### PR TITLE
[Bugfix:SubminiPolls] Prevent Quote Escaping in Histogram

### LINKED ISSUE #11256: Putting strings in quotes results in display of escaped characters in polling interface
Putting the strings in double quotes (making the numbers strings)
Screenshot 2024-11-01 at 11 53 03 PM

results in display of escaped characters : `&quot;`
Screenshot 2024-11-01 at 11 52 33 PM

_Originally posted by @bmcutler in https://github.com/Submitty/Submitty/issues/11115#issuecomment-2452844438_

### PR DESCRIPTION
**Pull request by bhavyaBeliever**
* [ ] Tests for the changes have been added/updated (if possible)
   * [ ] Documentation has been updated/added if relevant
   * [x] Screenshots are attached to Github PR if visual/UI changes were made
Closes #11256 
   ### What is the current behavior?

   If the x-labels of a Histogram contain strings of numbers, the quotes in the strings are being represented as `&quot;`  instead of the expected `"` format.
 This issue arises due to automatic escaping, which converts quotes to their HTML entity representation.

 ### What is the new behavior?
 The x-labels of the Histogram now correctly display strings of numbers with quotes represented as `"` (double quotes)
 instead of `&quot;`.

Screenshot from 2024-12-19 17-04-57

### CODE REVIEW

File: site/app/templates/polls/ViewPoll.twig [hunk 1]
```diff
@@ -169,8 +169,8 @@
             [window.getComputedStyle(document.body).getPropertyValue('--histogram-correct'),
                 window.getComputedStyle(document.body).getPropertyValue('--histogram-default')];
             {% for option in poll.getOptions() %}
-                data[0].x.push("{{ option.getResponse()|replace({"\n": " ", "\r": " ", "\t": " "}) }}");
-                data[0].y.push({{ response_counts[option.getId()] }});
+                data[0].x.push("{{ option.getResponse()|raw |escape('js') |replace({"\n": " ", "\r": " ", "\t": " "}) }}");
```
Comment (cjreed121) [inline site/app/templates/polls/ViewPoll.twig:172]: Does this work without the `raw`? Can also shorten `escape('js')` to `e('js')`

File: site/app/templates/polls/ViewPoll.twig [hunk 2]
```diff
@@ -169,8 +169,8 @@
...
+                data[0].y.push({{ response_counts[ option.getId()] }});
```
Comment (cjreed121) [inline site/app/templates/polls/ViewPoll.twig:173]: unnecessary change. please revert

**Review by cjreed121 — CHANGES_REQUESTED** [review]

File: site/app/templates/polls/ViewPoll.twig (hunk 1 above)
Comment (bhavyaBeliever) [reply site/app/templates/polls/ViewPoll.twig:172]: Thanks @cjreed121 for the suggestion.
Yeah, after researching and testing I found we actually don't need `raw` for our application
